#164691202 Unit tests for existing functionality

richien opened the pull request:
#### What does this PR do?

Add unit tests to existing functionality in the codebase

#### Description of Task to be completed?

Write tests for the following features

`models`
`views`
`serializers`

#### How should this be manually tested?

- After pulling the branch to your local repo, change into it using 
 `git checkout <branch-name>`  and run `python manage.py test`

#### Any background context you want to provide?

- Ensure you have a virtual environment setup with the packages in
 the `requirements.txt` file before attempting to run the tests.
 This will ensure all dependencies are met.

#### What are the relevant pivotal tracker stories?

[#164691202](https://www.pivotaltracker.com/story/show/164691202)

richien committed:
chore(unittests): Add unittests for user model and user manager

- check that the user model can create a user
- check that the user manager can create a regular user
- check that the user manager can create a super user

[#164691202]

richien committed:
chore(unittests): Integrate django-nose and coverage reports

- Install django-nose and coverage
- Integrate django-nose to run the tests and return coverage reports

[164691202]

richien committed:
chore(unittests): Add tests for the user registration endpoint

- Test for registration of unregistered username and email
- Test for registration attempt with existing username and email
- Test for registration attempt with password less than eight
  characters

[#164691202]

richien committed:
chore(unittests) Add stub for unittests of the RetrieveUpdate view

- Add stub for testing the retrieving and updating a user to be
implemented later

[#164691202]

richien committed:
chore(TravisCI): Integrate TravisCI for test automation on builds

- Add configuration for TravisCI to .travis.yml file
- Remove .coveragerc from lit of files in .gitignore

[164691209]

Manorlds-Eaglespark committed:
chore(upgrade): upgrade project to django 2.1
- ensure the project shell is using the latest version of django at the time of development
Finishes 164691204

richien committed:
chore(unittests): Test endpoints that retrieve and update a user

- Test that only an authenticated user can retrieve or update their
data

[#164691202]

Manorlds-Eaglespark committed:
chore(upgrade): upgrade project to django 2.1
- add requirements.txt content
Finishes 164691204

richien committed:
chore(Upgrade Django): Merge 'ch-add-unittests-164691202' into ch-upgrade-to-django2.1

- Add unit tests to the upgraded version of the application

richien committed:
chore(Upgrade to Django-2.1): Refractor tests

- Append forward-slash to api URI
- Resolve merge conflicts

reifred reviewed (approved):
@richien I have tried out the tests and everything is working out fine. The coverage so far is also good.

richien committed:
chore(unittests): Add tests for serializers

- Check that the LoginSerializer can validate a user
- Check that the LoginSerializer returns a dictionary
with expected user fields

[Finishes #164691202]

AmosWels reviewed (commented):
Good work @richien, kindly squash your commits into one commit. I see they are all applied to the same task.

Also endeavor to follow the team's code conventions of writing a commit message. You have to add `Delivers` if its a chore, `Finishes` for a feature or `Fixes` for a bug. Please visit https://github.com/aqua-organisation/ah-django-Aqua/wiki/Coding-Conventions-for-ah-django-Aqua

richien commented:
> 
> 
> Good work @richien, kindly squash your commits into one commit. I see they are all applied to the same task.
> 
> Also endeavor to follow the team's code conventions of writing a commit message. You have to add `Delivers` if its a chore, `Finishes` for a feature or `Fixes` for a bug. Please visit https://github.com/aqua-organisation/ah-django-Aqua/wiki/Coding-Conventions-for-ah-django-Aqua

Thank you @AmosWels  for the feedback. I will make the necessary corrections as you have requested.

richien committed:
chore(Integrate TravisCI): Add Travis build badge to README.md

- Add TravisCI badge to README.md with default branch as develop

[DELIVERS #164691209]

richien committed:
Merge branch 'ch-Integrate-TravisCI-164691209' into ch-add-unittests-164691202
